### PR TITLE
Md feature/9/create new reaction

### LINKED ISSUE #9: Create New Reaction
As an admin I would like to be able to add a new Reaction so that I can increase the variety of Reactions available to users of the system.

### PR DESCRIPTION
**Pull request by Michael-Deitz**
### **Purpose**

To allow the user with the role Admin to see a list of all Reactions and the ability to create a Reaction.

### **Files Changed**

- Created file, and added the class with the needed properties in new `ReactionCreateDTO.cs`
- Created file, and added the endpoint for getting all Reactions in new `ReactionController.cs`
- Added the endpoint for creating a Reaction in `ReactionController.cs`
- Created file, and added the fetch for getting all Reactions in new `ReactionManager.js`
- Added the fetch for creating a Reaction in `ReactionManager.js`
- Created file, and wrote initial rendering and logic for Reaction Management view in new `ReactionList.jsx`
- Created file, and wrote initial rendering and logic for Creating a Reaction in new `CreateReaction.jsx`
- Added routing for Reaction Management in `ApplicationView.jsx`
- Added nav button to Reaction Management in `NavBar.jsx`
- Added routing for Creating a Reaction in `ApplicationView.jsx`

### **To Test**

Fetch, setup and run according to project README, then confirm the following 

- Logged in Admin can see a nav button for the Reactions Management view in the navbar
- Logged in Admin can navigate to the Reactions Management view and see a list of all Reactions 
- Logged in User with out the Admin role should not be able to see the Reactions Management view in the navbar
- Logged in Admin can click Create A Reaction and get navigated to a form to fill out
- Logged in Admin can click the Create button and create a Reaction
- Logged in Admin gets navigated to the Reactions Management view

closes #9

### CODE REVIEW

File: client/src/components/reactions/CreateReaction.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            <h4 className="mt-2" style={{display: 'flex', justifyContent: 'center'}}>Create a Reaction</h4>
+            <Card className="w-75 shadow" outline color="light" style={{maxWidth: "1200px"}}>
+                <CardBody>
+                    <Form className="w-50 m-auto"
+                        style={{maxWidth:"20rem"}}
+                        onSubmit={handleSubmit}
+                    >
+                        <FormGroup>
+                            <Label className="fw-bold">Reaction Name</Label>
+                                <Input
```
Comment (ztrouy) [inline client/src/components/reactions/CreateReaction.jsx:33]: Unindent line 33-40 once

File: client/src/components/reactions/CreateReaction.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                                    value={name}
+                                    placeholder="Enter Name For The Reaction"
+                                    onChange={(e) => {
+                                    setName(e.target.value);
+                                    }}
+                                />
+                        </FormGroup> 
+                        <FormGroup>
+                            <Label className="fw-bold">Reaction Image</Label>
+                                <Input
```
Comment (ztrouy) [inline client/src/components/reactions/CreateReaction.jsx:44]: Unident line 44-51 once

File: client/src/components/reactions/CreateReaction.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                        </FormGroup>
+                        <ButtonToolbar className="d-flex justify-content-end gap-2">
+                            <Button type="submit">Create</Button>
+                        </ButtonToolbar>
+                    </Form>
+                </CardBody>
+            </Card>
+        </PageContainer>
+    )
+
```
Comment (ztrouy) [inline client/src/components/reactions/CreateReaction.jsx:61]: Remove the whitespace line 61

File: Controllers/ReactionController.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    [HttpPost]
+    [Authorize(Roles = "Admin")]
+    public IActionResult ReactionCreate(ReactionCreateDTO newReaction)
+    {
+        Reaction reactionToCreate = new Reaction()
+        {
+            Name = newReaction.Name,
+            ReactionImage = newReaction.ReactionImage
+        };
+        _dbContext.Reactions.Add(reactionToCreate);
```
Comment (ztrouy) [inline Controllers/ReactionController.cs:45]: Add a line of whitespace between line 44 and 45

**Review by ccote95 — APPROVED** [review]

File: Controllers/ReactionController.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        _dbContext.SaveChanges();
+        return Created($"/api/reaction/{reactionToCreate.Id}", reactionToCreate);
```
Comment (ztrouy) [inline Controllers/ReactionController.cs:47]: Add a line of whitespace between `_dbContext.SaveChanges()` and the return

File: client/src/components/reactions/ReactionList.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    }, [])
+
+    return (
+        <>
+            <h4 style={{display: 'flex', justifyContent: 'center'}}>Reactions</h4>
+                <div>
+                    <div style={{ display: 'flex', justifyContent: 'center', marginBottom: '1rem' }}>
+                        <Button color="success" onClick={() => navigate("/reactions/create")}>Create a Reaction</Button>
+                    </div>
+                {reactions.map((r) => (
```
Comment (ztrouy) [inline client/src/components/reactions/ReactionList.jsx:22]: Indent line 22-30 by one

**Review by ztrouy — CHANGES_REQUESTED** [review]

**Review by ztrouy — APPROVED** [review]
:shipit:

**Pull request merged**